lib/ansible/modules/network/vyos/vyos_vlan.py

```diff
@@ -0,0 +1,294 @@
+            if not obj_in_want:
+                for i in h['interfaces']:
+                    commands.append('delete interfaces ethernet {0} vif {1}'.format(i, h['vlan_id']))
+
+    return commands
+
+
+def map_params_to_obj(module):
+    obj = []
+    aggregate = module.params.get('aggregate')
+    if aggregate:
+        for item in aggregate:
+            for key in item:
+                if item.get(key) is None:
+                    item[key] = module.params[key]
+
+            d = item.copy()
+            d['vlan_id'] = str(d['vlan_id'])
+
+            obj.append(d)
+    else:
+        obj.append({
+            'vlan_id': str(module.params['vlan_id']),
+            'name': module.params['name'],
+            'address': module.params['address'],
+            'state': module.params['state'],
+            'interfaces': module.params['interfaces']
+        })
+
+    return obj
+
+
+def map_config_to_obj(module):
+    objs = []
+    interfaces = list()
+
+    output = run_commands(module, 'show interfaces')
+    lines = output[0].strip().splitlines()[3:]
+
+    for l in lines:
+        splitted_line = re.split(r'\s{2,}', l.strip())
+        obj = {}
+
+        eth = splitted_line[0].strip("'")
+        if eth.startswith('eth'):
+            obj['interfaces'] = []
+            if '.' in eth:
+                interface = eth.split('.')[0]
+                obj['interfaces'].append(interface)
+                obj['vlan_id'] = eth.split('.')[-1]
+            else:
+                obj['interfaces'].append(eth)
+                obj['vlan_id'] = None
+
+            if splitted_line[1].strip("'") != '-':
+                obj['address'] = splitted_line[1].strip("'")
+
+            if len(splitted_line) > 3:
+                obj['name'] = splitted_line[3].strip("'")
+            obj['state'] = 'present'
+            objs.append(obj)
+
+    return objs
+
+
+def check_declarative_intent_params(want, module):
+    if module.params['interfaces']:
+        time.sleep(module.params['delay'])
+        have = map_config_to_obj(module)
+
+        for w in want:
+            for i in w['interfaces']:
+                obj_in_have = search_obj_in_list(w['vlan_id'], have)
+
+                if obj_in_have and 'interfaces' in obj_in_have and i not in obj_in_have['interfaces']:
+                    module.fail_json(msg='Interface {0} not configured on vlan {1}'.format(i, want['vlan_id']))
+
+
+def main():
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        vlan_id=dict(type='int', required=True),
+        name=dict(),
+        address=dict(),
+        interfaces=dict(type='list', required=True),
+        delay=dict(default=10, type='int'),
+        state=dict(default='present',
+                   choices=['present', 'absent'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['vlan_id'] = dict(required=True)
+    aggregate_spec['interfaces'] = dict(required=True)
```

Comment:
Why `dict(required=True)` is added here? It is already part of the spec.

Reply:
Looks like I forgot to remove this. This was added before I added the logic in element_spec. Thanks for pointing out.